lib/system_win32.c

```diff
@@ -267,4 +267,17 @@ HMODULE Curl_load_library(LPCTSTR filename)
 #endif
 }
 
+bool Curl_win32_impersonating(void)
+{
+  bool ret = FALSE;
+#ifndef CURL_WINDOWS_APP
+  HANDLE token = NULL;
+  if(OpenThreadToken(GetCurrentThread(), TOKEN_QUERY, TRUE, &token)) {
+    CloseHandle(token);
+    ret = TRUE;
+  }
+#endif
```

Comment:
Perhaps, this code should also be enabled for winapp:

```suggestion
  HANDLE token = NULL;
  if(OpenThreadToken(GetCurrentThread(), TOKEN_QUERY, TRUE, &token)) {
    CloseHandle(token);
    ret = TRUE;
  }
```